com.unity.render-pipelines.high-definition/Editor/Lighting/LightUnit/LightUnitSlider.cs

```diff
@@ -0,0 +1,316 @@
+            float min = rect.x;
+            float max = (rect.x + rect.width) - width;
+            markerRect.x = Mathf.Clamp(markerRect.x, min, max);
+
+            // Draw marker by manually drawing the rect, and an empty label with the tooltip.
+            EditorGUI.DrawRect(markerRect, Color.white);
+
+            // Scale the marker tooltip for easier discovery
+            const float markerTooltipRectScale = SliderConfig.k_MarkerTooltipScale;
+            var markerTooltipRect = markerRect;
+            markerTooltipRect.width  *= markerTooltipRectScale;
+            markerTooltipRect.height *= markerTooltipRectScale;
+            markerTooltipRect.x      -= (markerTooltipRect.width  * 0.5f) - 1;
+            markerTooltipRect.y      -= (markerTooltipRect.height * 0.5f) - 1;
+            EditorGUI.LabelField(markerTooltipRect, GetLightUnitTooltip(tooltip, value));
+        }
+
+        void DoIcon(Rect rect, GUIContent icon, float range)
+        {
+            var oldColor = GUI.color;
+            GUI.color = Color.clear;
+            EditorGUI.DrawTextureTransparent(rect, icon.image);
+            GUI.color = oldColor;
+
+            EditorGUI.LabelField(rect, GetLightUnitTooltip(icon.tooltip, range));
+        }
+
+        void DoThumbTooltip(Rect rect, float position, float value, string tooltip)
+        {
+            const float size = SliderConfig.k_ThumbTooltipSize;
+            const float halfSize = SliderConfig.k_ThumbTooltipSize * 0.5f;
+
+            var thumbMarkerRect = rect;
+            thumbMarkerRect.width  = size;
+            thumbMarkerRect.height = size;
+
+            // Vertically align with slider
+            thumbMarkerRect.y += halfSize - 1f;
+
+            // Horizontally place tooltip on the wheel,
+            thumbMarkerRect.x  = rect.x + (rect.width - size) * position;
+
+            EditorGUI.LabelField(thumbMarkerRect, GetLightUnitTooltip(tooltip, value));
+        }
+
+        GUIContent GetLightUnitTooltip(string baseTooltip, float value)
+        {
+            string formatValue;
+
+            if (value >= 100000)
+                formatValue = (value / 1000).ToString("#,0K");
+            else if (value >= 10000)
+                formatValue = (value / 1000).ToString("0.#") + "K";
+            else
+                formatValue = value.ToString("#.0");
+
+            string tooltip = baseTooltip + " - " + formatValue + " " + m_Descriptor.unitName;
+
+            return new GUIContent(string.Empty, tooltip);
+        }
+
+        protected virtual void DoSlider(Rect rect, SerializedProperty value, Vector2 range)
+        {
+            value.floatValue = GUI.HorizontalSlider(rect, value.floatValue, range.x, range.y, GUI.skin.horizontalSlider, GUI.skin.horizontalSliderThumb);
+        }
+
+        static float Remap(float v, float x0, float y0, float x1 = 0f, float y1 = 1f) => x1 + (v - x0) * (y1 - x1) / (y0 - x0);
+
+        protected virtual float GetPositionOnSlider(float value)
```

Comment:
A small comment on protected API can be nice to know what is expected in the implementation.